test/end2end.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright 2012-2016 Ghent University
+#
+# This file is part of vsc-mympirun,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-mympirun
+#
+# vsc-mympirun is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-mympirun is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-mympirun.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+End-to-end tests for mympirun (with mocking of real 'mpirun' command).
+
+@author: Jeroen De Clerck (HPC-UGent)
+@author: Kenneth Hoste (HPC-UGent)
+@author: Caroline De Brouwer (HPC-UGent)
+"""
+import copy
+import os
+import re
+import shutil
+import stat
+import tempfile
+import unittest
+from vsc.utils.run import run_simple
+
+
+FAKE_MPIRUN = """#!/bin/bash
+echo 'fake mpirun'
+echo $@
```

Comment:
do a single echo like:

```
echo "fake mpirun called with args: $@"
```

then you can check more strictly to make sure the fake `mpirun` is really called